src/Internal/Service/Service.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+declare(strict_types=1); // @codeCoverageIgnore
+
+namespace Rinq\Internal\Service;
+
+/**
+ * Service is an interface for background tasks that can finish with an error.
+ */
+interface Service
```

Comment:
I don't know yet, but I don't think this interface will be necessary in PHP. 
In Go it's used to wait for several sub-systems that are running in separate threads.